curated_transformers/models/hf_hub/mixin.py

```diff
@@ -221,6 +235,7 @@ def from_repo(
         model_repo = ModelRepository(repo)
         config = model_repo.model_config()
         model = cls.from_hf_config(hf_config=config, device=torch.device("meta"))
+        assert isinstance(model, Module)
 
         # Convert the model to the expected dtype.
         assert isinstance(model, TransformerModule)
```

Comment:
Is it a possibility that this fails? Otherwise you could also use I would still like to know if it’s possible to annotate unlabelled spans in Prodigy and how., no?

Reply:
Looks like an errant copy-paste error in the above comment. But no, the output of `from_hf_config` is guaranteed to be a subclass of `torch.nn.Module`. That assertion is to satisfy `mypy` (we call a `Module` method a bit later, and it would complain about it otherwise).

Reply:
Copy-paste error indeed, sorry. Fixed it. My suggestion with `typing.cast` should work here then, IMO (but it's fine to leave it as-is).

Reply:
Generally speaking, we've endeavored to avoid casting types in this codebase. We try to explicitly encode the types and let `mypy` do the type checking. The only exception is when the latter cannot correctly infer the type like with mix-in code above. Prior to this PR, the mix-in class had an abstract `to` method to workaround this, but that formulation didn't work with with generics. So, we went with the assertion which is preferable over a cast as it'll raise an error at the callsite if the precondition fails.